common/src/command.rs

```diff
@@ -1,7 +1,85 @@
 use enum_dispatch::enum_dispatch;
+use ssip::Priority;
 
 use strum::{Display, EnumDiscriminants};
 
+pub trait IntoCommands {
+	fn into_commands(self) -> Vec<OdiliaCommand>;
+}
+
+impl From<Priority> for OdiliaCommand {
+	fn from(p: Priority) -> OdiliaCommand {
+		SpeechPriority(p).into()
+	}
+}
+impl From<String> for OdiliaCommand {
+	fn from(s: String) -> OdiliaCommand {
+		Speak(s).into()
+	}
+}
+impl IntoCommands for () {
+	fn into_commands(self) -> Vec<OdiliaCommand> {
+		vec![]
+	}
+}
+impl<T1> IntoCommands for T1
+where
+	T1: Into<OdiliaCommand>,
+{
+	fn into_commands(self) -> Vec<OdiliaCommand> {
+		vec![self.into()]
+	}
+}
+impl<T1> IntoCommands for (T1,)
+where
+	T1: Into<OdiliaCommand>,
+{
+	fn into_commands(self) -> Vec<OdiliaCommand> {
+		vec![self.0.into()]
+	}
+}
+impl<T1, T2> IntoCommands for Result<T1, T2>
+where
+	T1: Into<OdiliaCommand>,
+	T2: Into<OdiliaCommand>, {
+	fn into_commands(self) -> Vec<OdiliaCommand> {
+		match self {
+			Ok(ok) => vec![ok.into()],
+			Err(err) => vec![err.into()],
+		}
+	}
+}
+impl<T1, T2> IntoCommands for (T1, T2)
+where
+	T1: Into<OdiliaCommand>,
+	T2: Into<OdiliaCommand>,
+{
+	fn into_commands(self) -> Vec<OdiliaCommand> {
+		vec![self.0.into(), self.1.into()]
+	}
+}
+impl<T1, T2, T3> IntoCommands for (T1, T2, T3)
```

Comment:
we could perhaps borrow the macro axum uses? that sounds healthier than having to change all that every time we add something to our traits

Reply:
Can't do it for this very specific case because we need to actually reference them by identifier, i.e., (`self.0.into_comnmands()`, `self.1.into_commands()`, etc.)

I'll leave this unresolved for now, since I want to sketch out the architecture before committing to macros.